src/App.tsx

```diff
@@ -1,24 +1,148 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import { UserWarning } from './UserWarning';
 
-const USER_ID = 0;
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+
+import { ErrorOnPage } from './components/ErrorOnPage';
+
+import { FilterBy } from './utils/FilterBy';
+import { Todo } from './types/Todo';
+import { getTodos, createTodo, removeTodo } from './todos';
+
+import { getFilteredTodos } from './utils/NewFilterTodos';
+
+import { ErrorMessages } from './types/ErrorNessages';
+
+const USER_ID = 1333;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.All);
+  const [newError, setNewError] = useState<ErrorMessages | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [deleteTodosId, setDeleteTodosId] = useState<number[]>([]);
+
+  const isTodoShow = todos.length > 0;
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => {
+        setNewError(ErrorMessages.LoadError);
+      });
+  }, []);
+
+  const deleteTodo = async (todoId: number) => {
+    try {
+      setDeleteTodosId((prev) => [...prev, todoId]);
+
+      const removeNewTodo = await removeTodo(todoId);
+
+      if (!removeNewTodo) {
+        setNewError(ErrorMessages.DeleteError);
+
+        return;
+      }
+
+      setTodos((prev) => {
+        const filterPrev = prev.filter((todo) => todo.id !== todoId);
+
+        return [...filterPrev];
+      });
+    } catch (error) {
+      setNewError(ErrorMessages.DeleteError);
+    } finally {
+      setDeleteTodosId([]);
+    }
+  };
+
+  const removeCompletTodos = (todosDelete: Todo[]) => {
+    todosDelete.forEach(async (todo) => {
+      await deleteTodo(todo.id);
+    });
+  };
+
+  const filterTodos = useMemo(() => {
+    return getFilteredTodos(todos, filterBy);
+  }, [todos, filterBy]);
+
+  const handleNewTodoTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodoTitle(event.target.value);
+  };
+
+  const clearTodoTitle = () => {
+    setNewTodoTitle('');
+  };
+
+  const addNewTodo = async (title: string) => {
+    if (!title) {
```

Comment:
Return type of async functions should be Promise. Please define the return type of addNewTodo function as Promise<null | Todo>